### ISSUE TITLE
Interaction has already been acknowleged directly after using interaction.reply or interaction.deferReply

### ISSUE THREAD

**Issue by fheahdythdr** [discordjs/discord.js#9052]
### Which package is this bug report for?

discord.js

### Issue description

I'm not sure how to reproduce it, because it just started happening randomly.
![image](https://user-images.githubusercontent.com/100938045/212131533-6673ff01-927f-410d-9aec-084ed8cddb2d.png)
These are the logs after I remove await interaction.deferReply().
Moment I add await interaction.deferReply() it does this. 
![image](https://user-images.githubusercontent.com/100938045/212131735-7fa4c28b-79d0-4e80-9be3-b09fe5e1b1ce.png)

Code submitted is with await interaction.deferReply(). Happens with all of my other bots too.

### Code sample

```typescript
client.on('interactionCreate', async interaction => {
    if (!interaction.isCommand()) return;
    await interaction.deferReply()
    const command = client.commands.get(interaction.commandName);
    if (!command) return;
    try {
        await command.execute(interaction, client);
    } catch (error) {
        if (error) console.error(error);
        
        if (!interaction.replied && !interaction.deferred) {
            await interaction.reply(`There was an error while executing this command, error is ${error}`);
        }
        else if (interaction.replied || interaction.deferred) {
            await interaction.editReply(`There was an error while executing this command, error is ${error}`);
        }   
    }
});
```

### Package version

14.6.0

### Node.js version

v19.3.0

### Operating system

Manjaro Linux

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

User, Message

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildVoiceStates, GuildPresences, MessageContent

**Comment by fheahdythdr** [comment]
Looking at it, it might be executing the callback twice, because replacing await interaction.deferReply() with await interaction.reply('Processing.') also makes it error with Interaction has already been acknowledged while still having the reply go through.

**Comment by fheahdythdr** [comment]
Actually, I'm pretty sure it's .deferReply() and .reply() being called twice for some reason.

**Comment by kyranet** [comment]
This is because `deferReply` sends a response to `/callback`, and later sets state:

https://github.com/discordjs/discord.js/blob/05a1cbfe5479195b0bc9b6f0971fe39f6af6fd77/packages/discord.js/src/structures/interfaces/InteractionResponses.js#L66-L81

If you `reply` to an interaction that was either deferred or replied, you'll get an error:

https://github.com/discordjs/discord.js/blob/05a1cbfe5479195b0bc9b6f0971fe39f6af6fd77/packages/discord.js/src/structures/interfaces/InteractionResponses.js#L101-L102

That's why, if you use `deferReply`, all subsequent calls must use `editReply` instead:

https://github.com/discordjs/discord.js/blob/05a1cbfe5479195b0bc9b6f0971fe39f6af6fd77/packages/discord.js/src/structures/interfaces/InteractionResponses.js#L156-L157

---

Basically, you have already sent a reply, and you're sending more replies, rather than updating the initial one. That's why `deferReply` or `reply` before another `reply` will make the second one fail.

**Comment by fheahdythdr** [comment]
All of my commands use editReply() though. The error happens before I even call command.execute (in the client.on('interactionCreate')).

As in, the event gets fired, and it'll error at the exact line I call deferReply() (which i call before any other reply functions)

**Comment by fheahdythdr** [comment]
Nevermind, everything's just being executed twice.
![image](https://user-images.githubusercontent.com/100938045/212337337-ae762fa3-b48e-413d-adb2-20f24f5d52e9.png)
This was from one command, which is supposed to only add one card to the Trello page, code below.

`
        const modal = new ModalBuilder()
        .setTitle('Submit a suggestion.')
        .setCustomId(interaction.user.id + 'Suggest')

        const titlet = new TextInputBuilder()
        .setLabel('Title of suggestion')
        .setStyle(TextInputStyle.Short)
        .setCustomId('title')

        const desct = new TextInputBuilder()
        .setLabel('Description of suggestion')
        .setStyle(TextInputStyle.Paragraph)
        .setCustomId('desc')

        const titleac = new ActionRowBuilder().addComponents(titlet)
        const descac = new ActionRowBuilder().addComponents(desct)

        modal.addComponents(titleac, descac)

        try {await interaction.showModal(modal);} catch{}
        async function thing(i) {
            if (!i.isModalSubmit()) return;
            const title = i.fields.getTextInputValue("title");
            const desc = (i.fields.getTextInputValue("desc") || "No description provided.") + "\n\nAdded by: " + interaction.user.tag + "\n\n\n\n\n\nDiscord ID (For pipedream workflow):" + interaction.user.id;
            trello.addCard(title, desc, "my list", async () => {
                try {await i.reply({content: "Added card to Trello list. View it here: link to thing", ephemeral: true})} catch {}
            })
            client.removeListener('interactionCreate', thing)
        }
        client.on('interactionCreate', thing)
`

**Comment by fheahdythdr** [comment]
The try{} catch{} lines with the reply and showModal are there because it decides to execute twice.

**Comment by fheahdythdr** [comment]
The debug logs don't help much either, as they stop once I use a command (for some reason)

**Comment by fheahdythdr** [comment]
It's just executing things twice, looking at it.
![image](https://user-images.githubusercontent.com/100938045/212344505-ad49426b-431c-4522-b2a5-9afd633d7008.png)
Here's the code.
````js
        const modal = new ModalBuilder()
        .setTitle('Submit a suggestion.')
        .setCustomId(interaction.user.id + 'Suggest')

        const titlet = new TextInputBuilder()
        .setLabel('Title of suggestion')
        .setStyle(TextInputStyle.Short)
        .setCustomId('title')

        const desct = new TextInputBuilder()
        .setLabel('Description of suggestion')
        .setStyle(TextInputStyle.Paragraph)
        .setCustomId('desc')

        const titleac = new ActionRowBuilder().addComponents(titlet)
        const descac = new ActionRowBuilder().addComponents(desct)

        modal.addComponents(titleac, descac)

        try {await interaction.showModal(modal);} catch{}
        async function thing(i) {
            if (!i.isModalSubmit()) return;
            const title = i.fields.getTextInputValue("title");
            const desc = (i.fields.getTextInputValue("desc") || "No description provided.") + "\n\nAdded by: " + interaction.user.tag + "\n\n\n\n\n\nDiscord ID (For pipedream workflow):" + interaction.user.id;
            trello.addCard(title, desc, "my list id", async () => {
                try {await i.reply({content: "Added card to Trello list. View it here: my trello page", ephemeral: true})} catch {}
            })
            client.removeListener('interactionCreate', thing)
        }
        client.on('interactionCreate', thing)
```
This is supposed to add one card to the Trello page, but it adds two for some reason.

**Comment by fheahdythdr** [comment]
Debug logs don't help either, as they stop once I execute a command (I don't know why)

**Comment by kyranet** [comment]
Your code is way too complex, please write a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example).

**Comment by fheahdythdr** [comment]
I'm not sure how to replicate it, so it might just be an issue on my end.

**Comment by fheahdythdr** [comment]
Alright I just decided to hack together a solution and modified the code responsible for the error (in @discordjs and discord.js folders in node_modules) lmao
Think it's just an issue on my end.
